dartagnan/src/main/java/com/dat3m/dartagnan/program/event/arch/tso/Xchg.java

```diff
@@ -32,8 +33,8 @@ public Register getResultRegister(){
     }
 
     @Override
-    public ImmutableSet<Register> getDataRegs(){
-        return ImmutableSet.of(resultRegister);
+    public Set<Register.Read> getRegisterReads(){
+        return Set.of(new Register.Read(resultRegister, Register.UsageType.DATA));
```

Comment:
why data and not address? we are simply exchanging the values in two addresses, aren't we?

Reply:
From what I understood is that this event exchanges the data between an address and a register (hence the reg is a data dependency).
But you are right that the address dependencies are not getting collected.